libyul/backends/evm/EVMCodeTransform.cpp

```diff
@@ -563,6 +563,10 @@ void CodeTransform::operator()(Block const& _block)
 	Scope* originalScope = m_scope;
 	m_scope = m_info.scopes.at(&_block).get();
 
+	for (auto const& statement: _block.statements)
+		if (auto function = get_if<FunctionDefinition>(&statement))
+			createFunctionEntryID(*function);
```

Comment:
Should the order of function definitions in the block matter? Keeping the same order as the source would be nice. This visit would change it for functions defined inside a nested block.

Reply:
Previously, it depended on the order in which functions were called, now it depends on the order in which they are defined (in a kind of breadth-first-search), I think this is better.

Reply:
Not sure if it matters, function hoister and function grouper would change the visit order in that case. Wouldn't it be nice if the order was the same? Maybe doing this in `operator()(FunctionDefinition const&...)` is more appropriate?

Also, where else are using the current order?

Reply:
It cannot be done in `operator()(FunctionDefinition` because we need the entry id to be available before the function can be called for the first time, that's the point of this change. I'm fine with the internal label numbers to change (note that this has zero effect on the generated bytecode) as an effect of the function hoister or grouper.